azurelinuxagent/common/protocol/restapi.py

```diff
@@ -178,6 +179,14 @@ def __init__(self, name=None):
         self.properties = ExtHandlerProperties()
         self.versionUris = DataContractList(ExtHandlerVersionUri)
 
+    def sort_key(self):
+        level = self.properties.dependencyLevel
+        if level is None:
+            level = 0
+        if self.properties.state != u"enabled":
+            level = (0 - level) - 1
```

Comment:
Would be good to have a comment here. I didn't understand the reason for (0 - level) -1.